GVFS/GVFS.Platform.Windows/ProjFSFilter.cs

```diff
@@ -286,6 +281,90 @@ public static bool IsNativeLibInstalled(ITracer tracer, PhysicalFileSystem fileS
             return existsInSystem32 || existsInAppDirectory;
         }
 
+        public static bool TryCopyNativeLibIfDriverVersionsMatch(ITracer tracer, PhysicalFileSystem fileSystem, out string copyNativeDllError)
+        {
+            string system32NativeLibraryPath = Path.Combine(Environment.SystemDirectory, ProjFSNativeLibFileName);
+            if (fileSystem.FileExists(system32NativeLibraryPath))
+            {
+                copyNativeDllError = $"{ProjFSNativeLibFileName} already exists at {system32NativeLibraryPath}";
+                return false;
+            }
+
+            string gvfsProcessLocation = ProcessHelper.GetCurrentProcessLocation();
+            string appDirectoryNativeLibraryPath;
+            string installedNativeLibraryPath;
+            GetNativeLibPaths(gvfsProcessLocation, out installedNativeLibraryPath, out appDirectoryNativeLibraryPath);
+            if (fileSystem.FileExists(appDirectoryNativeLibraryPath))
+            {
+                copyNativeDllError = $"{ProjFSNativeLibFileName} already exists at {appDirectoryNativeLibraryPath}";
+                return false;
+            }
+
+            if (!fileSystem.FileExists(installedNativeLibraryPath))
+            {
+                copyNativeDllError = $"{installedNativeLibraryPath} not found, no {ProjFSNativeLibFileName} available to copy";
+                return false;
+            }
+
+            string installedPrjfltDriverPath = Path.Combine(gvfsProcessLocation, "Filter", DriverFileName);
+            if (!fileSystem.FileExists(installedPrjfltDriverPath))
+            {
+                copyNativeDllError = $"{installedPrjfltDriverPath} not found, unable to validate that packaged driver matches installed driver";
+                return false;
+            }
+
+            string system32PrjfltDriverPath = Path.Combine(Environment.ExpandEnvironmentVariables(System32DriversRoot), DriverFileName);
+            if (!fileSystem.FileExists(system32PrjfltDriverPath))
+            {
+                copyNativeDllError = $"{system32PrjfltDriverPath} not found, unable to validate that packaged driver matches installed driver";
+                return false;
+            }
+
+            FileVersionInfo packagedDriverVersion;
+            FileVersionInfo system32DriverVersion;
+            try
+            {
+                packagedDriverVersion = FileVersionInfo.GetVersionInfo(installedPrjfltDriverPath);
```

Comment:
Should these methods exist on `PhysicalFileSystem` to make this method more testable?

Reply:
Next commit will add this to `PhysicalFileSystem`, and I will look into adding unit test(s) in a subsequent commit.